### DISCUSSION TITLE
Yubikey Support

### DISCUSSION THREAD

**Discussion by Prolonz** [ente-io/ente#502] [Enhancements]
It would be great to add support for FIDO/passkeys so accounts can be secured with physical security keys like Yubikey.

I love the idea of an open source cross platform 2FA service but am hesitant to use it if it’s (somewhat ironically) only able to be secured by a password.

**Comment by LearningAsIGo71** [comment]
Good point... It certainly is ironic! A 2FA app that can be logged into without 2FA (unless you choose email 2FA).

**Comment by Kilobyte4621** [comment]
I'm on the same boat here.. I love the value proposition of a FOSS Authenticator app with e2ee multi-device syncing, but I'm also hesitant secure all my 2fa tokens just behind a password in a system that's exposed to the internet.. FIDO2/Passkey support would allow me to use a hardware security key and still have the protection factor of "something I have", which is the whole purpose of 2fa codes in the first place. Gonna keep an eye on it to see if it gets implemented, and definitely gonna switch to it once it happens.

**Comment by ghost** [comment]
I use ente auth as a secondary TOTP provider. For most of my TOTP, I just use my password manager to fill it in automatically. But if/when I need to enter a TOTP for my password manager, then I rely on ente auth. That makes the security level of ente auth critical. 

I'm all for it too.

**Comment by Metr0pl3x** [comment]
Possible integration for this outside of using Play Services akin to: https://gitlab.com/kunzisoft/android-hardware-key-driver?

**Comment by mnvr** [comment]
This is done now - Passkey support, including for Yubikey and other hardware keys, has been added. For more details, please see https://ente.io/blog/introducing-passkeys-on-ente/.

**Reply by Kilobyte4621** [reply]
That's awesome! Thanks a lot for the effort! Definitely going to give it a try :dancers:
